### PR TITLE
Change subscript to throw on missing key

### PR DESCRIPTION
**Pull request by cberner**
Change map subscript to throw an exception for keys that do not exist in
the map. The old behavior can be acheived with the element_at function.

### CODE REVIEW

**Comment by cberner** [comment]
I'll wait to merge this until the release after `element_at` is merged for maps

**Comment by martint** [comment]
looks good

**Comment by cberner** [comment]
@martint, I added a config flag to ease migration for this (which I'll squash before merging). Could you take a look at that last commit?

File: presto-main/src/main/java/com/facebook/presto/operator/scalar/MapSubscriptOperator.java [hunk 1]
```diff
@@ -98,7 +99,7 @@ else if (keyType.getJavaType() == Slice.class) {
     }
 
     @UsedByGeneratedCode
-    public static Object subscript(MethodHandle keyEqualsMethod, Type keyType, Type valueType, Block map, boolean key)
+    public static Object subscript(Boolean legacyMissingKey, MethodHandle keyEqualsMethod, Type keyType, Type valueType, Block map, boolean key)
```
Comment (martint) [inline presto-main/src/main/java/com/facebook/presto/operator/scalar/MapSubscriptOperator.java]: Can't this use primitive boolean?

Comment (cberner) [inline presto-main/src/main/java/com/facebook/presto/operator/scalar/MapSubscriptOperator.java]: No, `bindTo` only works on reference types

**Pull request merged**